### DISCUSSION TITLE
What is the latest stable version of this project?

### DISCUSSION THREAD

**Discussion by cbhiii** [ESPresense/ESPresense#293] [Q&A]
I came to know about the project after seeing it featured on [Everything Smart Home](https://youtu.be/p7C2QvmsM8M). 

Seeing as it was listed as version 2. x software I would have thought it to be a much more mature and stable release than what I'm experiencing. **I don't want to sound critical, because I appreciate all the time and work that has gone into this project and I lack the skills to implement and maintain such an undertaking.** However, in carrying a 2.x label, I would think that there would be a stable updates channel available vs. what seem like nightly updates at this point.

Does anyone know what the last stable working version of this project was? And how I can get it loaded into my units until the latest issues (rolling reboots, non-reporting of updates) are worked out?

I love the idea of this project and want it to succeed. If you're a newcomer and start using it like I did, you might be surprised at the issues being encountered. I'm just looking for more stability right now so I can begin to see how I would want to implement and use this information in Home Assistant. As it is not, I can't get that far because it's not working as it should. Thank you.

**Comment by FreakyBoy** [comment]
I have made a downgrade to v2.0.25. Everything runs without reboots and the distance is also displayed in home assistant.

**Reply by DeekB1983** [reply]
How do you downgrade?

I could not find a way to choose firmware from https://espresense.com/

**Comment by stevencoutts** [comment]
Download from https://github.com/ESPresense/ESPresense/releases <https://github.com/ESPresense/ESPresense/releases>

Flash manually.

**Reply by DeekB1983** [reply]
Thanks, will have a look into flashing the bin file manually, never tried it before always used the web based flasher.

**Reply by erwinSmith757** [reply]
im very interested in the same, what is stable and what is not stable? I also learned about this amazing project through same youtube channel .... immediately i went to setup a esp in each room down stairs ...and automations, but then quickly realised its very very unstable....i would like to experiment on this in a semi stable release....but every update made it more unstable to be honest ...

i would recommend releases are labeled ( stable / experimental / etc ) 

but cant complain this is amazing work!!!!!!!

**Reply by ASNNetworks** [reply]
Thanks, I see that there have been a few updates the last few days. I downgraded a few versions lower since I have been experiencing issues for the last week (somewhere end last week also). Just to make sure it works like it used to before this.

**Comment by DeekB1983** [comment]
I agree with above, I was running V2.0.24 and it was the most stable so far but upgraded to V2.0.27 which seems worse off, going to try downgrading to .24 or .25 as .27 is not working well for me.

**Comment by cbhiii** [comment]
I manually installed v2.0.24 and unchecked auto update via the esphome-flasher software. I'll see how this goes. Thanks for the info.

**Comment by cbhiii** [comment]
Unfortunately it seems that v2.0.24 does not work reliably when you populate the query field. I had the field blank before, but then it stopped seeing my Apple watch after a short while.

Does anyone know of a stable version of this project before v2.0.24?

I have the sensors deployed across my house and I'm trying to learn how they work so I can include them in automations, but my Apple watch can be right next to the sensor (1 meter) and then it just stops reporting or seeing it and then is marked 'not_home'.

**Reply by ASNNetworks** [reply]
I downgrades to 2.0.15. All releases after that started a week ago, same time as my issues started. https://github.com/ESPresense/ESPresense/releases/tag/v2.0.15

This is the last version before the update "spree" and I had no issues before last week. So I was running this release before then.

**Reply by DeekB1983** [reply]
> Unfortunately it seems that v2.0.24 does not work reliably when you populate the query field. I had the field blank before, but then it stopped seeing my Apple watch after a short while.
> 
> Does anyone know of a stable version of this project before v2.0.24?
> 
> I have the sensors deployed across my house and I'm trying to learn how they work so I can include them in automations, but my Apple watch can be right next to the sensor (1 meter) and then it just stops reporting or seeing it and then is marked 'not_home'.

I have using V2.0.24 and the query function was working reliably for me, I am now on V2.0.25 and that also seem to work reliably for the query function.

I was having the same issue with the Iphones I am tracking going to not home, it looks like the phone stops broadcasting (Guessing its going into a sleep state) I got round this by extending my away timeout to 720 seconds (made things slightly better on the iphones I am tracking)
![image](https://user-images.githubusercontent.com/64811870/154544444-0cc2ef77-cbc3-457b-b248-6b619c6d6ce6.png)

I have been unable to tack any apple watch successfully, they always seem to go to not_home after 10 - 20 minutes and never come back unless disconnecting and reconnecting Bluetooth, tracking apple devices as a whole has been somewhat of a challenge for me and looking at other posts a lot of other people seem to have issues tracking apple devices, my wife and daughter are using apple devices moslty and they have been a pain for me to track correctly, I logged a github request on the IOS Home Assistant section to include a BLE TRasnmittir like they have done for android in the hope that it brings me some stability - https://github.com/home-assistant/iOS/discussions/2081

I do have great success tracking my android devices using the HA BLE Transmitter function from the companion app and a miband 6, both on V2.0.24 and V2.0.25 and this is how I have the query set
![image](https://user-images.githubusercontent.com/64811870/154545793-b3fb872c-cb78-4a59-90b1-906f7886df02.png)

and looking in mqtt I see the devices I have set up for query appearing which makes it look like that functionality is working
![image](https://user-images.githubusercontent.com/64811870/154545971-4ae3f8e6-cb17-4752-b4bd-dffbf4474e46.png)

I think the issue you be facing is more around the apple devices rather than the query function as they seem to stop broadcasting frequently and you can see this my watching the terminal which is not great I agree, I am hoping some future release brings Iphone stability. 


Go have a vote on it although it wont help with the apple watch tracking but I am hoping having it increases Iphone stability,

**Reply by cbhiii** [reply]
Thank you. This is great detail and background since I'm new coming to the project. You could be right in that this relates to more of an apple issue. I'll review the settings in detail this evening. Thanks again.

**Reply by k3vyn32** [reply]
I use the Room Assistant App on iPhones to transmit  (I have now migrated away from Room Assistant, but still use the app.)
https://apps.apple.com/gb/app/room-assistant/id1538642237
Looks like this in MQTT Explorer -
<img width="713" alt="Screenshot 2022-02-17 at 19 06 30" src="https://user-images.githubusercontent.com/59889969/154552792-f90035c1-3565-4bb2-b3e5-79742299ac58.png">

**Reply by ctschach** [reply]
> I use the Room Assistant App on iPhones to transmit (I have now migrated away from Room Assistant, but still use the app.)
> https://apps.apple.com/gb/app/room-assistant/id1538642237
> Looks like this in MQTT Explorer -
> <img alt="Screenshot 2022-02-17 at 19 06 30" width="713" src="https://user-images.githubusercontent.com/59889969/154552792-f90035c1-3565-4bb2-b3e5-79742299ac58.png">

Stupid question if you say you have moved away from room-assistant.

meaning you use the home assistant companion app plus ESPresence  together? Or do you use any additional tools?

Anybody know of something similar which will run on the Apple Watch?

**Reply by k3vyn32** [reply]
I use the Room Assistant App to give a consistent Bluetooth address from each iPhone.
and then use an esp32 in each room to track that address.

**Reply by DeekB1983** [reply]
> I use the Room Assistant App on iPhones to transmit (I have now migrated away from Room Assistant, but still use the app.) https://apps.apple.com/gb/app/room-assistant/id1538642237 Looks like this in MQTT Explorer - <img alt="Screenshot 2022-02-17 at 19 06 30" width="713" src="https://user-images.githubusercontent.com/59889969/154552792-f90035c1-3565-4bb2-b3e5-79742299ac58.png">

@k3vyn32 I am also using the room assistant app but have very little success with it, I seem to have multiple Ids for the same devices too I think.

Look at my mqtt screenshot, pay attention to the amount of messages
![image](https://user-images.githubusercontent.com/64811870/154796134-fbc16a2d-2488-4a19-b97d-98b5f9eb3bab.png)

The items marked in green are my wifes Iphone 13, this is now being tracked successfully under the both ID's (Room assistant app is installed but I never receive any messages under the room assistant ID) You can see its regularly checking in and has many messages in MQTT
![image](https://user-images.githubusercontent.com/64811870/154796209-04a1157f-9040-43b6-a8de-76e22d85f67f.png)

In the red is my daughters Iphone 11 on the other hand, again this has the room assistant app installed with the same setting, I receive very little messages from room assistant, the other ID's in red are the same phone but it varies the amount of messages coming through to each ID and you can see lots of not_home activity
![image](https://user-images.githubusercontent.com/64811870/154796407-f5ac18ae-155e-48ac-9e02-f116a85fa8cd.png)

My daughter was home last night from 8PM through to this morning, her phone was charging all night but you can see the large not_home area where no messages were being received into mqtt, my wifes phone was doing the same but has started working all of a sudden, I have been through the aIl the phone app settings and changed the room assistant access to Always but not sure if there are other settings that should be changed.

Also the 1005 id's are my daughters I-watch, you can see very little messages from them and its not checked in even although its here all the time with her phone which is not the most reliable at being tracked either
![image](https://user-images.githubusercontent.com/64811870/154796513-50b62fe3-b481-4719-94ea-52105bddc2b4.png)

Also it looks like the ID's coming in from the same iPhones swap between whatever ID'#s, I see the same for my wifes Iphone but getting regular messages in both topics, if you look at the MQ now you can see the topic apple:12 is has not received just as more messages in mqtt than the room assistant topic has
![image](https://user-images.githubusercontent.com/64811870/154796762-a83409f0-4eef-4f19-b5f4-4403d8db62fa.png)

The messages seem to come regular if my daughter is using her phone, if she puts it down for a period of inactivity the messages stop coming in like the app is going into a sleep state although its set to always allow which has been challenging for me

**Comment by cbhiii** [comment]
Thanks @ASNNetworks. I'll give that a shot.

**Reply by DeekB1983** [reply]
I also use the room assistant app if you see my query filter above but have limited auccess sometimes.

Also for people looking to track iwatches should check this  thread out https://github.com/ESPresense/ESPresense/issues/231

**Comment by cbhiii** [comment]
I stumbled upon this project is anyone is looking for just Apple Watch detection. Seems to work really well if your focus is limited to just Apple Watch. [ESPHome Apple Watch detection](https://github.com/dalehumby/ESPHome-Apple-Watch-detection)

**Comment by DTTerastar** [comment]
https://github.com/ESPresense/ESPresense/discussions/317
